### DISCUSSION TITLE
Rust nightly support (on diagnostic check)

### DISCUSSION THREAD

**Discussion by dcow** [zed-industries/zed#6507] [Feature Requests]
Is there a way to generally configure zed to use rust nightly instead of mainline rust when determining diagnostic notes to display? We have a nightly feature in use and so we constantly have a little issue in the bottom left:
<img width="80" alt="Screenshot 2023-03-20 at 5 18 57 PM" src="https://user-images.githubusercontent.com/547926/226486260-110301ae-f83f-4788-9ed2-03d267c860d8.png">

<img width="582" alt="Screenshot 2023-03-20 at 5 21 18 PM" src="https://user-images.githubusercontent.com/547926/226486548-e0e7c7a9-8b86-454f-b41c-b51a8cfe871e.png">

**Comment by lukasnor** [comment]
You can configure the LSP settings that Zed uses. It is described [here](https://zed.dev/docs/configuring-zed#lsp). You can set the the rust-analyzer to use nightly as described [here](https://rust-analyzer.github.io/manual.html#toolchain).
So this does the trick 
```json
{
  "lsp": {
    "rust-analyzer": {
      "initialization_options": {
        "server": {
          "extraEnv": {
            "RUSTUP_TOOLCHAIN": "nightly"
          }
        }
      }
    }
  }
}
```

**Reply by nixpulvis** [reply]
I have a `rust-toolchain.toml` file, why isn't it being used?

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links

**Reply by nixpulvis** [reply]
Just my 2 cents, but Discord is harder to search through existing issues, so I would assume it creates more duplicate issues. But if your team is more comfortable talking there I get it. I just wish it wasn't locked behind a login server.
